src/role_nextroomer.js

```diff
@@ -308,11 +308,15 @@ roles.nextroomer.settle = function(creep) {
     methods.push(Creep.constructTask);
   }
 
+  // Fill energy after building is done
+  if (creep.room.controller.level >= 4) {
+    methods.push(Creep.transferEnergy);
+  }
```

Comment:
I think I found a solution that would base filling upon how many parts the creep has.
If the nextroomer would be only as efficient as a sourcer it has been better in my testing to be building sourcers and having carries deliver to the nextroomers to keep more time on upgrading the controller.

 ```
 // Fill energy after building is done
  if (creep.getActiveBodyparts('work') <= 5 && creep.room.controller.level >= 3) {
    methods.push(Creep.transferEnergy);
  }
```